src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs

```diff
@@ -184,45 +203,27 @@ private static FieldDeclaration CreateFieldFromExisting(ISymbol existingMember,
             CodeWriterDeclaration declaration = new CodeWriterDeclaration(existingMember.Name);
             declaration.SetActualName(existingMember.Name);
 
-            // TODO -- come up a way to get the value type from customized code
-            // when creating field from existing member (customized code), here we use the same type as its value type.
-            // for 2 reasons:
-            // 1. it works
-            // 2. we do not really have an efficiant way to get the value type of a property because we really cannot relying on the information in the schema when you have to do a customization, and from the csharp code, there is a lot of information we cannot get.
-            return new FieldDeclaration($"Must be removed by post-generation processing,", fieldModifiers, fieldType, fieldType, declaration, GetPropertyDefaultValue(originalType, inputModelProperty), inputModelProperty.IsRequired, inputModelProperty.SerializationFormat, existingMember is IFieldSymbol, writeAsProperty, SerializationMapping: serialization);
+            return new FieldDeclaration($"Must be removed by post-generation processing,", fieldModifiers, fieldType, valueType, declaration, GetPropertyDefaultValue(originalType, inputModelProperty), inputModelProperty.IsRequired, inputModelProperty.SerializationFormat, existingMember is IFieldSymbol, writeAsProperty, SerializationMapping: serialization);
         }
 
-        private static INamedTypeSymbol GetExistingType(ISymbol existingMember) => existingMember switch
-        {
-            IPropertySymbol propertySymbol => (INamedTypeSymbol)propertySymbol.Type,
-            IFieldSymbol fieldSymbol => (INamedTypeSymbol)fieldSymbol.Type,
-            _ => throw new NotSupportedException($"'{existingMember.ContainingType.Name}.{existingMember.Name}' must be either field or property.")
-        };
-
         private static bool IsReadOnly(ISymbol existingMember) => existingMember switch
         {
             IPropertySymbol propertySymbol => propertySymbol.SetMethod == null,
             IFieldSymbol fieldSymbol => fieldSymbol.IsReadOnly,
             _ => throw new NotSupportedException($"'{existingMember.ContainingType.Name}.{existingMember.Name}' must be either field or property.")
         };
 
-        private static (CSharpType PropertyType, CSharpType ValueType) GetPropertyDefaultType(in InputModelTypeUsage modelUsage, in InputModelProperty property, TypeFactory typeFactory)
+        private static CSharpType GetPropertyDefaultType(in InputModelTypeUsage usage, in InputModelProperty property, TypeFactory typeFactory)
         {
             var propertyType = typeFactory.CreateType(property.Type);
 
-            if (modelUsage == InputModelTypeUsage.Output ||
+            if (!usage.HasFlag(InputModelTypeUsage.Input) ||
```

Comment:
Why change usage from `Output` to `Input` here?

Reply:
the usage is a "flag", and actually it has two unique flags - Input and Output.
RoundTrip is a combination of input and output.

The condition `usage == Output` is the same as `!usage.HasFlag(Input)`. Iterate all the cases to find out:
| usage value | result of `== Output` | result of `!HasFlag(Input) |
| :-- | :-- | :-- |
| None | false | false |
| Input | false | false |
| Output | true | true |
| RoundTrip | false | false |